src/App.tsx

```diff
@@ -1,61 +1,111 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
 
-export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
 
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+  return foundUser || null;
+}
 
-          <span className="error">Please choose a user</span>
-        </div>
+function getLargestTodoId(todos: Todo[]) {
+  return Math.max(...todos.map(todo => todo.id));
+}
 
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [newTodoUserId, setNewTodoUserId] = useState('');
+  const [todosState, setTodos] = useState(todos);
+
+  const [isTitleError, setIsTitleError] = useState(false);
```

Comment:
```suggestion
  const [isTitleError, setTitleError] = useState(false);
```

Reply:
be consistent: either `setUserError setTitleError` or `setIsUserError setIsTitleError`

for boolean variable names is better to use ` isUserError setUserError `

Reply:
Fixed, thank you